Wikipedia/Categories/NSString+Extras.m

```diff
@@ -144,4 +142,17 @@ - (NSString*)wmf_trim {
     return [self stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
 }
 
+- (NSString*)wmf_substringUpToString:(NSString*)string{
+    return [[self componentsSeparatedByString:string] firstObject];
+}
+- (NSString*)wmf_substringAfterString:(NSString*)string{
+   NSArray* components = [self componentsSeparatedByString:string];
```

Comment:
I don't think this would work for cases like `[@"foo1bar1baz" wmf_substringAfterString:@"1"]`. I would expect it to return `@"1bar1baz"` but seems like this would return `@"baz"`, is that intentional?  If not, I would consider using `rangeOfString:` and `substring(To|From)Index:`.